yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -258,7 +258,8 @@ private[spark] class Client(
     if (executorMem > maxMem) {
       throw new IllegalArgumentException(s"Required executor memory (${args.executorMemory}" +
         s"+$executorMemoryOverhead MB) is above the max threshold ($maxMem MB) of this cluster! " +
-        "Please increase the value of 'yarn.scheduler.maximum-allocation-mb'.")
+        "Please increase the value of 'yarn.scheduler.maximum-allocation-mb' and" +
```

Comment:
"and/or"? it could really be due to either one -- the max the scheduler will schedule, or the limit on the max size of a container. It may not need an increase in both.